File: src/my_project/easy_problems/from101to150/peak_index_mountain_array.py
from typing import List 

class SecondSolution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:

        arr_copy: List[int] = arr.copy()
        arr_copy.sort()
        arr_max: int = arr_copy[-1]



        return arr.index(arr_max)



class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:

        max: int = -1 
        solution: int = 0
        for i in range(len(arr)):
            if arr[i] > max:
                max = arr[i]
                solution = i 
        return solution 


solution = Solution()
print(solution.peakIndexInMountainArray(arr = [0,2,1,0]))
print(solution.peakIndexInMountainArray(arr = [0,10,5,2]))
print(solution.peakIndexInMountainArray(arr = [3,4,5,1]))

File: my_project_typescript/easy_problems_typescript/from1to50/peak_index_mountain_array.ts
function peakIndexInMountainArray(arr: number[]): number {

    let solution: number = 0;
    let max: number = -1;

    arr.map( (x, index) => {
        if (x > max){
            max = x
            solution = index
        }
    } )

    return solution
};

console.log((peakIndexInMountainArray([24,69,100,99,79,78,67,36,26,19])))